### PR TITLE
[JENKINS-28550] Unwanted effect rendering the pages

### PR DESCRIPTION
**Pull request by recena**
In the ticket description it can see a screencast showing the problem. More information: https://issues.jenkins-ci.org/browse/JENKINS-28550

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
There was a timeout on the build system. Restarted the build.

**Comment by recena** [comment]
@oleg-nenashev Thanks so much. This is my first PR to Jenkins project. Did you see the screenscast?

**Comment by KostyaSha** [comment]
@recena did you successfully run tests locally?

**Comment by recena** [comment]
@KostyaSha I found the problem. In CI env there is ```-Dmaven.test.failure.ignore=true``` and I didn't see the problem. Thanks so much.

**Comment by daniel-beck** [comment]
Triggering another PR build…

**Comment by recena** [comment]
@daniel-beck Thanks.

**Comment by daniel-beck** [comment]
The sidepanel is variable width again, being ridiculously wide on large screens. And the layout scripts that took @tfennelly several iterations to get working nicely don't appear to work until after the first AJAX update. This is a huge step backward for something as insignificant as layout moving around while the page is loading.

**Comment by recena** [comment]
@daniel-beck Do we want a side-panel with fixed-size? Ok, no problem but we should do using Bootstrap framework not Javascript.

**Comment by daniel-beck** [comment]
Well, the current implementation is fixed size, and this changes it unintentionally (or I'd have to ask why you didn't mention this in the opening post). And I think fixed width is the correct way to go, especially given the advanced handling of build history items we now have.

**Comment by recena** [comment]
@daniel-beck The side-panel came back to be fixed-width. Really, the side-panel has two different sizes: *320px* if your screen has <=1024px (width), and *360px* (original size) if your screen has >1024px

**Comment by daniel-beck** [comment]
@recena Doesn't work properly. See screen shot.

**Comment by aheritier** [comment]
I reproduced the problem @daniel-beck has thus :-1: for the current PR. Otherwise I'm clearly in favour of such change to not use javascript to setup the layout when everything is loaded (which gives the strange effect) and to use a fixed with for the left column.

**Comment by recena** [comment]
@daniel-beck @aheritier Thanks for your feedback. I've solved the problem. There is much work around to Jenkins UI.

**Comment by daniel-beck** [comment]
@recena Did you also fix the AJAX update issue I showed you?

**Comment by recena** [comment]
@daniel-beck No, I did because I thought doing in other PR (with other ticket). What do you think about it?

**Comment by jtnord** [comment]
BTW - someone did some work so that this actually works on mobile sized screen (so don't forget about that),  and the build queue and executor status should fill the whole window when in "that" mode.

image

**Comment by jtnord** [comment]
@daniel-beck the current implementation is not fixed size, try narrowing your window, you will see the "mobile" type layout and in that mode the side panel will gracefully resize to the width of your window.

**Comment by recena** [comment]
@jtnord This is opposite to "we want a fixed-left menu".

**Comment by jtnord** [comment]
@recena - exactly - I think others may have forgotten about the small screen work as they live on hi-res all day :-P

Now - the current small-screen layout could be improved (does it have to show exactly the same as the full view, is the side panel (build queue etc) more important that the Jobs?), so maybe it is worth taking a step back and thinking about the bigger picture - so the hi-res can have a fixed width, but a mobile view has something completely different.

**Comment by recena** [comment]
@daniel-beck @jtnord I attached some screenshots with the result (640px, 1024px and 1280px):

jenkins-28550_640px
jenkins-28550_1024px
jenkins-28550_1280px

**Comment by jtnord** [comment]
screenshots look ok - and the code changes look reasonable.  
That said I have not built and tested this lcoally, nor am I a CSS expert.

This has the potential to break page decorators that are looking for the divs with the IDs that are now removed.
A quick [search](https://github.com/search?l=&q=%22side-panel-content%22+user%3Ajenkinsci&ref=advsearch&type=Code&utf8=%E2%9C%93) on github for [side-panel-content](https://github.com/search?l=&q=%22side-panel-content%22+user%3Ajenkinsci&ref=advsearch&type=Code&utf8=%E2%9C%93) showed [timestamper](https://github.com/jenkinsci/timestamper-plugin/blob/f733398baa2735cd9f89dfb717bfc940c9af4cef/src/main/resources/hudson/plugins/timestamper/annotator/TimestampAnnotatorFactory/script.js#L138) would be affected and likewise a search for [main-panel-content](https://github.com/search?l=&q=%22main-panel-content%22+user%3Ajenkinsci&ref=advsearch&type=Code&utf8=%E2%9C%93) showed the [acceptance test harness](https://github.com/jenkinsci/acceptance-test-harness/blob/e56bb4f408a95efbfca950cca0740942b6d77ac9/src/main/java/org/jenkinsci/test/acceptance/plugins/emma/EmmaResultsPage.java#L54) is affected and both will need follow up PRs.

**Comment by recena** [comment]
@jtnord If you agree, I'm going to resolve [this](https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/2780/testReport/jenkins.widgets/BuildListTableTest/linksFromFolders) first. And then I'll review the rest of tests. The easy way would be to add ``div```, but this PR tries to improve some things.

**Comment by daniel-beck** [comment]
@jtnord 

> the current implementation is not fixed size

Well, it is except for the mobile view. The first version of this PR still had the sidepanel grow ridiculously wide, and it happened on every screen resize.

**Comment by daniel-beck** [comment]
Triggering another PR build...

@recena So far I don't see why we should have a different width sidepanel on 1024 than on 1280. Currently the sidepanel is the same width from ~800 to 2560px wide. What is the reason for this change?

**Comment by recena** [comment]
@daniel-beck IMO, the side-panel in screen's width of 1024px is too big (360px). I like a bit more small (320px) in order to use the space in a better way.

**Comment by daniel-beck** [comment]
Makes sense. So tentative :+1: from me once the layouting delay is fixed (will test badges and all the weird stuff again once that's changed to make sure it didn't break).

**Comment by daniel-beck** [comment]
@tfennelly's input may also be helpful. He probably knows the pitfalls best.

**Comment by recena** [comment]
@daniel-beck I've solved the inherited problem from set a fixed side-panel (based on CSS, not JS)

**Comment by daniel-beck** [comment]
Compare the following two screenshots. One is Jenkins 1.617, the other is PR build 2785.

Previously, if the build name wasn't too long (e.g. just the build number), the build date and time were placed on the same line (@tfennelly may know the details better — he worked quite hard to get these details right). Now, the build name seems to more quickly get moved to the next line. Additionally, the badge of build `#2` seems to be word wrapped unnecessarily.



PR build

Please note that I'm not rejecting this PR – I'm not sure about the relative merits, maybe the original issue warrants a minor change in behavior – just pointing out that a previously deliberate behavior changed in this PR.

**Comment by recena** [comment]
@daniel-beck Thanks so much for your feedback. I'm going to analyze these screenshots and I'll back with results. On the other hand, I would like to say:

- [x] IMO, this PR's motivation seems logical. Don't forget this [render effect](https://drive.google.com/file/d/0B0sHMazJqTbmT3JnQzlkcGhYUFE/view?usp=sharing).
- [x] This PR removes code and simplifies things.
- [x] If we know the UI layout, why rebuild it with JS? Result: strange effects
- [x] The last revisions in this PR are related to side effects

**Comment by recena** [comment]
@daniel-beck I updated my forked repository and took this screenshot:

jenkins-pr-1717

What is wrong?

**Comment by recena** [comment]
@daniel-beck Another screenshot with badges included by **Build Trigger Badge Plugin**.

jenkins-pr-1717_2

**Comment by recena** [comment]
@daniel-beck Another one including the stop button.

jenkins-pr-1717_3

**Comment by recena** [comment]
@daniel-beck @tfennelly Any comment?

**Comment by recena** [comment]
@reviewbybees Could anyone give me more feedback? Thanks in advance.

File: core/src/main/resources/lib/layout/layout.jelly [hunk 1]
```diff
@@ -227,52 +227,45 @@ ${h.initPageVariables(context)}
...
-          <st:include it="${pd}" page="footer.jelly" optional="true" />
-        </j:forEach>
-        </j:if>
       </div>
-    </div>
+    </footer>
+
   </body>
 </html>
 </j:jelly>
```
Comment (tfennelly) [inline core/src/main/resources/lib/layout/layout.jelly]: Hey Manuel, it's hard for me to see what has changed here. It looks as though a lot of the changes were markup formatting (non-functional) changes, no?

File: war/src/main/webapp/css/style.css [hunk 2]
```diff
@@ -75,68 +75,77 @@ body {
 }
 
 #side-panel {
-  z-index: 1;
```
Comment (tfennelly) [inline war/src/main/webapp/css/style.css]: `z-index` was added for a reason. I can't remember why atm. Is it causing an issue of some form?

**Comment by tfennelly** [comment]
@recena hey Manuel. Give me a chance to play with this before it gets pushed.

**Comment by olivergondza** [comment]
:+1:

**Comment by tfennelly** [comment]
:+1: 

@recena Thanks for fixing this.

File: war/src/main/webapp/css/style.css (hunk 2 above)
Comment (recena) [inline war/src/main/webapp/css/style.css]: This proposal does not need to manage the overlap between elements.

**Comment by recena** [comment]
@olivergondza @tfennelly Thanks so much for your feedback.

I'm going to review this [test](https://github.com/jenkinsci/acceptance-test-harness/blob/e56bb4f408a95efbfca950cca0740942b6d77ac9/src/main/java/org/jenkinsci/test/acceptance/plugins/emma/EmmaResultsPage.java#L54) (acceptance test harness).

File: core/src/main/resources/lib/layout/layout.jelly (hunk 1 above)
Comment (recena) [inline core/src/main/resources/lib/layout/layout.jelly]: Sure, It is not easy to see the changes. Basically, I've tried to simplify the markup.

**Comment by amuniz** [comment]
:bee:

**Comment by daniel-beck** [comment]
:bee: for the current PR build.

However this PR should not be merged as requested by @recena while he ensures this does not break the ATH.

**Comment by recena** [comment]
@daniel-beck I'll do it. Thanks.

**Comment by recena** [comment]
@daniel-beck The test has been reviewed. Finally, if Jenkins Core committers decide to merge it, I'll be alert to solve possible bugs.

**Comment by daniel-beck** [comment]
This may have caused [JENKINS-29250](https://issues.jenkins-ci.org/browse/JENKINS-29250).

**Comment by recena** [comment]
@daniel-beck Working on it.

**Comment by recena** [comment]
@daniel-beck [PR](https://github.com/jenkinsci/sectioned-view-plugin/pull/8) sent.

**Comment by recena** [comment]
@bpedersen2 Have you filled a Jira issue? I'm sure we can find a quick solutions.

**Pull request merged**